pytorch_lightning/cluster_environments/slurm_environment.py

```diff
@@ -25,9 +25,10 @@ def __init__(self):
 
     def master_address(self):
         # figure out the root node addr
-        try:
-            root_node = os.environ["SLURM_NODELIST"].split(" ")[0]
-        except Exception:
+        slurm_nodelist = os.environ.get("SLURM_NODELIST")
```

Comment:
There is a SLURM refactor already at https://github.com/PyTorchLightning/pytorch-lightning/pull/5102

Reply:
ok, so let's see which one lands first :]

Reply:
I'd suggest not refactor here and just add the TODOs to be resolved in another PR